### PR TITLE
protocol/tx: Dereference the data entry when initializing a vmcontext refdata hash

### PR DESCRIPTION
**Pull request by bobg**

### CODE REVIEW

**Comment by bobg** [comment]
PTAL

**Comment by bobg** [comment]
PTAL

**Comment by bobg** [comment]
Here is a :cookie: PTAL

File: protocol/tx/transaction.go [hunk 1]
```diff
@@ -49,6 +49,18 @@ func TxHashes(oldTx *bc.TxData) (hashes *bc.TxHashes, err error) {
...
 	hashes.VMContexts = make([]*bc.VMContext, len(oldTx.Inputs))
 
+	getRefDataHash := func(id entryRef) bc.Hash {
+		dEntry, ok := entries[id]
+		if !ok {
+			return bc.EmptyStringHash
+		}
+		d, ok := dEntry.(*data)
+		if !ok {
+			return bc.EmptyStringHash
```
Comment (jbowens) [inline protocol/tx/transaction.go]: this shouldn't be an error?

Comment (bobg) [inline protocol/tx/transaction.go]: I suppose so, thanks. Fixed.

**Comment by jbowens** [comment]
lgtm. it might be worth trying it without the `getRefDataHash` closure to avoid extra indirection? it might even be the same number of lines of code.

**Pull request merged**